internal/provider/user_role_grant_resource_test.go

```diff
@@ -171,61 +198,80 @@ func TestIntegrationRoleGrantResource(t *testing.T) {
 		Return(nil, nil, nil)
 
-	// Post cleanup verification
-	s.EXPECT().GetAllRolesForUser(gomock.Any(), userID).
-		Return(&PreCreateGetResponse, nil, nil)
-
-	testRoleGrantResource(t, clusterName, false /* skipOrgTest */, true /* useMock */)
+	testRoleGrantResource(t, clusterName, true /* useMock */)
 }
 
-func testRoleGrantResource(t *testing.T, clusterName string, skipOrgTest, useMock bool) {
+func testRoleGrantResource(t *testing.T, clusterName string, useMock bool) {
 	var (
 		orgGrantResourceName     = "cockroach_user_role_grant.org_grant"
 		clusterGrantResourceName = "cockroach_user_role_grant.cluster_grant"
-		clusterResourceName      = "cockroach_cluster.test"
 	)
 	defer HookGlobal(&generateRandomPassword, func() (string, error) {
 		return testPassword, nil
 	})()
 
-	// compose checkFuncs so we can optionally exclude the org test.
 	checkFuncs := []resource.TestCheckFunc{
-		testRoleGrantExists(clusterGrantResourceName, clusterResourceName, string(client.ORGANIZATIONUSERROLETYPE_CLUSTER_OPERATOR_WRITER), string(client.FOLDERRESOURCETYPETYPE_CLUSTER)),
-	}
-	if !skipOrgTest {
-		checkFuncs = append(checkFuncs, testRoleGrantExists(orgGrantResourceName, clusterResourceName, string(client.ORGANIZATIONUSERROLETYPE_BILLING_COORDINATOR), string(client.RESOURCETYPETYPE_ORGANIZATION)))
+		testRoleGrantExists(clusterGrantResourceName, string(client.ORGANIZATIONUSERROLETYPE_CLUSTER_OPERATOR_WRITER), string(client.FOLDERRESOURCETYPETYPE_CLUSTER)),
+		testRoleGrantExists(orgGrantResourceName, string(client.ORGANIZATIONUSERROLETYPE_BILLING_COORDINATOR), string(client.RESOURCETYPETYPE_ORGANIZATION)),
 	}
 
 	resource.Test(t, resource.TestCase{
-		// After removal of the test resources, verify that the user running the
-		// tests still has at least one non ORG_MEMBER role to show that this
-		// deletion did not remove the other roles like the user_role_grants
-		// resource would.
-		CheckDestroy:             verifyRemainingRolesIntact(clusterResourceName),
 		IsUnitTest:               useMock,
 		PreCheck:                 func() { testAccPreCheck(t) },
 		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
 		Steps: []resource.TestStep{
 			{
-				Config: getTestRoleGrantResourceConfig(clusterName, skipOrgTest),
+				Config: getTestRoleGrantResourceConfig(clusterName, false),
 				Check:  resource.ComposeTestCheckFunc(checkFuncs...),
 			},
 			{
-				Config:      getTestRoleGrantResourceConfigUpdated(clusterName, skipOrgTest),
+				Config:      getTestRoleGrantResourceConfigUpdated(clusterName),
 				ExpectError: regexp.MustCompile(`Role already exists`),
 			},
+			// After removal of the org role, verify that the associated user still
+			// has the cluster role to show that this deletion did not remove the
+			// other roles like the user_role_grants resource would.
+			{
+				Config: getTestRoleGrantResourceConfig(clusterName, true),
+				Check:  resource.ComposeTestCheckFunc(checkFuncs[0]),
```

Comment:
It would improve readability to name that checkFunc before putting it in the array.  Or you could skip the array and just have something like:
```go
checkClusterRole := testRoleGrantExists(clusterGrantResourceName, string(client.ORGANIZATIONUSERROLETYPE_CLUSTER_OPERATOR_WRITER), string(client.FOLDERRESOURCETYPETYPE_CLUSTER))

checkOrgRole := testRoleGrantExists(orgGrantResourceName, string(client.ORGANIZATIONUSERROLETYPE_BILLING_COORDINATOR), string(client.RESOURCETYPETYPE_ORGANIZATION))

// ....

Check:  resource.ComposeTestCheckFunc(checkClusterRole, checkOrgRole)
```

Also, I don't think you need to use the `ComposeTestCheckFunc` when there is only one.  So this line would become:

```go
Check:  checkClusterRole,
```